### PR TITLE
feat!: Support Axios v1

### LINKED ISSUE #45: Axios v1.0.0 compatibility
Axios v1.0.0 got released a few days ago (see https://github.com/axios/axios/releases/tag/v1.0.0) but `axios-case-converter` only works with versions between >=0.23.0 and <1.0.0 unfortunately.

image

Thank you for creating this package, I found it very useful.

### PR DESCRIPTION
**Pull request by siketyan**
Closes #45 

> **Warning** BREAKING CHANGE: CreateAxiosRequestTransformer now must return AxiosRequestTransformer instead of AxiosResponseTransformer.

axios [1.0.0, 2.0.0) are now supported.
< 1.0.0 versions are also supported, but it will be deleted in the next major release.

To run jest testing on axios v1, I also added babel-jest to transform ES modules that axios exports because jest only supports CommonJS without enabling experimental ESM support implicitly.

### CODE REVIEW

File: jest.resolver.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module.exports = (path, options) => options.defaultResolver(path, {
+  ...options,
+  packageFilter: pkg =>
+    // HACK: Axios exports their module in both CommonJS style and ESM style.
+    //       However, jest does prefer the ESM one that occurs a runtime error.
+    //       So in this resolver the main module of axios package is replaced to CommonJS one.
+    //       This only applies to tests.
```
Comment (mpyw) [inline jest.resolver.js:7]: Nice solution 👍

File: src/types.ts [hunk 2]
```diff
@@ -78,7 +78,7 @@ export interface CreateAxiosInterceptor {
   (options?: AxiosCaseMiddlewareOptions): AxiosInterceptor;
 }
 export interface CreateAxiosRequestTransformer {
-  (options?: AxiosCaseMiddlewareOptions): AxiosResponseTransformer;
+  (options?: AxiosCaseMiddlewareOptions): AxiosRequestTransformer;
```
Comment (mpyw) [inline src/types.ts:81]: Nice catch ✨

**Review by mpyw — APPROVED** [review]

File: test/integration/basic.ts [hunk 3]
```diff
@@ -94,7 +94,7 @@ test('it should be converted on success', (done) => {
     })
     .then((response) => {
       expect(JSON.stringify(response.data)).toBe(JSON.stringify(camelData));
-      expect(response.headers.contentType).toBe('application/json');
+      expect(response.headers.contentType || response.headers['Content-Type']).toBe('application/json');
```
Comment (mpyw) [inline test/integration/basic.ts:97]: What's the purpose of these changes?

Comment (siketyan) [reply test/integration/basic.ts:97]: @mpyw `response.headers.contentType` is no longer available in axios >= 1.0, so I added `response.headers['Content-Type']` which actually exists. To keep backward compatibility, I left the previous one with logical OR. This mean this tests can be run even with axios < 1.0.

**Comment by siketyan** [comment]
@mpyw Could you please approve to run CI workflow for confirming the tests pass?

File: test/integration/basic.ts (hunk 3 above)
Comment (mpyw) [reply test/integration/basic.ts]: @siketyan 

Was `headers.contentType` really generated by `axios`, not by `axios-case-converter`? I suspect that the camelCase conversion no longer works for latest `axios`.

Comment (mpyw) [reply test/integration/basic.ts]: [axios v1.x bug: response header mutations via `transformResponse` are discarded · Issue #47 · mpyw/axios-case-converter](https://github.com/mpyw/axios-case-converter/issues/47)
